src/ui/menu.cpp

```diff
@@ -487,6 +487,22 @@ bool MenuBox::onProcessMessage(Message* msg)
   Menu* menu = MenuBox::getMenu();
 
   switch (msg->type()) {
+    case kCloseDisplayMessage:
+      if (menu)
+        menu->closeAll();
+
+      // In case that this event is received when we are inside a
+      // modal loop to show this menu box
+      // (showPopup()/openWindowInForeground()) we redirect this
+      // kCloseDisplayMessage to the main window.
+      if (window()->isForeground()) {
+        auto* closeMsg2 = new Message(msg->type());
```

Comment:
Great! Just a minor aesthetic suggestion. You could change:
`auto* closeMsg2 = new Message(msg->type());`
to
`auto* closeMsg2 = new Message(kCloseDisplayMessage);`
Just to avoid the idea that `msg->type()` could have a value other than `kCloseDisplayMessage`.